Lib/dataclasses.py

```diff
@@ -1164,6 +1164,13 @@ def _process_class(cls, init, repr, eq, order, unsafe_hash, frozen,
             # In some cases fetching a signature is not possible.
             # But, we surely should not fail in this case.
             text_sig = str(inspect.signature(cls)).replace(' -> None', '')
+        except NameError:
```

Comment:
Why not just start with SOURCE format? I don't think we need to catch NameError first.

Reply:
When we do that, a lot of tests fail with similar messages:

```python
    def test_docstring_two_fields(self):
        @dataclass
        class C:
            x: int
            y: int

        self.assertDocStrEqual(C.__doc__, "C(x:int, y:int)")
```

Produces:

```
======================================================================
FAIL: test_docstring_two_fields (test.test_dataclasses.TestDocString.test_docstring_two_fields)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/sobolev/Desktop/cpython2/Lib/test/test_dataclasses/__init__.py", line 2294, in test_docstring_two_fields
    self.assertDocStrEqual(C.__doc__, "C(x:int, y:int)")
    ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sobolev/Desktop/cpython2/Lib/test/test_dataclasses/__init__.py", line 2263, in assertDocStrEqual
    self.assertEqual(a.replace(' ', ''), b.replace(' ', ''))
    ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: "C(x:'__dataclass_type_x__',y:'__dataclas[46 chars]e__'" != 'C(x:int,y:int)'
- C(x:'__dataclass_type_x__',y:'__dataclass_type_y__')->'__dataclass___init___return_type__'
+ C(x:int,y:int)
```

I am not quite sure - why.